test/module/irohad/multi_sig_transactions/mst_processor_test.cpp

```diff
@@ -0,0 +1,262 @@
+  onPreparedTransactionsWrapper.subscribe();
+
+  auto onExpiredTransactionsWrapper = make_test_subscriber<CallExact>(
+      mst_processor->onExpiredTransactions(), 1);
+  onExpiredTransactionsWrapper.subscribe();
+
+  // ---------------------------------| when |----------------------------------
+  auto quorum = 1;
+  auto timeBefore = 110;
+  mst_processor->propagateTransaction(
+      makeTx("another hash", "sign", quorum, timeBefore));
+
+  // ---------------------------------| then |----------------------------------
+  ASSERT_TRUE(onStateUpdateWrapper.validate());
+  ASSERT_TRUE(onPreparedTransactionsWrapper.validate());
+  ASSERT_TRUE(onExpiredTransactionsWrapper.validate());
+}
+
+/**
+ * @given initialised mst processor
+ * AND our state contains one transactions TX with quorum 2
+ * AND wrappers on mst observables
+ *
+ * @when received new state from other peer via transport,
+ * that contains TX with another signature
+ *
+ * @then check that:
+ * state updated
+ * AND 1 prepared transaction (although quorum 1)
+ * AND 0 expired transactions
+ */
+TEST_F(MstProcessorTest, onUpdateFromTransportUsecase) {
+  // ---------------------------------| given |---------------------------------
+  auto onStateUpdateWrapper =
+      make_test_subscriber<CallExact>(mst_processor->onStateUpdate(), 1);
+  onStateUpdateWrapper.subscribe();
+
+  auto onPreparedTransactionsWrapper = make_test_subscriber<CallExact>(
+      mst_processor->onPreparedTransactions(), 1);
+  onPreparedTransactionsWrapper.subscribe();
+
+  auto onExpiredTransactionsWrapper = make_test_subscriber<CallExact>(
+      mst_processor->onExpiredTransactions(), 0);
+  onExpiredTransactionsWrapper.subscribe();
+
+  auto quorum = 2;
+  auto timeAfter = 112;
+  mst_processor->propagateTransaction(
+      makeTx("hash", "sign_one", quorum, timeAfter));
+
+  // ---------------------------------| when |----------------------------------
+  auto another_peer = makePeer("another", "another_pubkey");
+  auto transported_state = MstState::empty(make_shared<TestCompleter>());
+  transported_state += makeTx("hash", "sign_two", quorum, timeAfter);
+  mst_processor->onNewState(another_peer, transported_state);
+
+  // ---------------------------------| then |----------------------------------
+  ASSERT_TRUE(onStateUpdateWrapper.validate());
+  ASSERT_TRUE(onPreparedTransactionsWrapper.validate());
+  ASSERT_TRUE(onExpiredTransactionsWrapper.validate());
+}
+
+/**
+ * @given initialised mst processor
+ * AND our state contains one transaction
+ *
+ * @when received notification about new propagation
+ *
+ * @then check that:
+ * transport invoked for all peers
+ */
+
+TEST_F(MstProcessorTest, onNewPropagationUsecase) {
+  // ---------------------------------| given |---------------------------------
+  auto quorum = 2;
+  auto timeAfter = 112;
+  mst_processor->propagateTransaction(
+      makeTx("hash", "sign_one", quorum, timeAfter));
+  EXPECT_CALL(*transport, sendState(_, _)).Times(2);
```

Comment:
maybe fix to `.Times(quorum)`?

Reply:
This constant should be deduced, but not from quorum( this will be very confused), but from the size of `peers` vector.